### ISSUE TITLE
TO Go POST /api/1.3/servers database access error

### ISSUE THREAD

**Issue by dangogh** [apache/trafficcontrol#2682]
a `POST` to `/api/1.3/servers` results in a database access error if the profile indicated by the `profileId` has `cdn` entry that is null.  The `global` profile is one example.  Granted, a profile associated with a CDN should be required, but a sensible error message needs to be returned rather than "database access error".

The error in the traffic ops log is this:
```
ERROR: servers.go:128: could not scan cdnID from profile: sql: Scan error on column index 0: converting driver.Value type <nil> ("<nil>") to a int: invalid syntax
```

Here's a sample JSON for a server.  In this case,  profileId 1 is the global profile which has no cdn associated with it:
```
{
    "cachegroupId": 1,
    "cdnId": 1,
    "hostName": "ahost",
    "ipAddress": "192.168.1.99",
    "physLocationId": 1,
    "profileId": 1,
    "statusId": 6,
    "tcpPort": 443,
    "typeId": 19,
    "updPending": false,
    "interfaceName": "eth0",
    "domainName": "example.com",
    "interfaceMtu" : 3600,
    "ipNetmask" : "255.255.0.0",
    "ipGateway" : "192.168.1.1"
}
```

**Comment by dangogh** [comment]
alternatively,  perhaps that cdn column needs to be not null with the default being the id of the ALL cdn

**Comment by dangogh** [comment]
fixed by adding not null constraint #2711
